python/vision_explanation_methods/error_labeling/error_labeling.py

```diff
@@ -0,0 +1,248 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+"""Defines the Error Labeling Manager class."""
+
+from copy import deepcopy
+from enum import Enum
+
+import numpy as np
+import torchvision
+from torch import Tensor
+
+LABELS = 'labels'
+
+
+class ErrorLabelType(Enum):
+    """
+    Enum providing types of error labels.
+
+    If none, then the detection is not an error. It is a
+    correct prediction.
+    """
+
+    # the gt doesn't have a corresponding detection
+    MISSING = "missing"
+
+    # the model predicted detections, but there was nothing there
+    # this prediction must have a 0 iou score with all gt detections
+    BACKGROUND = "background"
+
+    # the predicted class is correct, bounding box does not have sufficient
+    # overlap with ground truth (based on the iou threshold)
+    LOCALIZATION = "localization"
+
+    # the predicted class is incorrect, the bounding box is correct
+    CLASS_NAME = "class_name"
+
+    # both the predicted class and bounding box are incorrect
+    CLASS_LOCALIZATION = "class_localization"
+
+    # the predicted class is correct, the bounding box is correct, but
+    # the iou score is lower than another detection
+    DUPLICATE_DETECTION = "duplicate_detection"
+
+    MATCH = "match"
+
+
+class ErrorLabeling():
+    """
+    Defines a wrapper class of Error Labeling for vision scenario.
+
+    Only supported for object detection at this point.
+    """
+
+    def __init__(self,
+                 task_type: str,
+                 pred_y: list,
+                 true_y: list,
+                 iou_threshold: float = 0.5):
+        """
+        Create an ErrorLabeling object.
+
+        :param task_type: The task to run.
+        :type task_type: str
+        :param pred_y: predicted detections, nested list of 6 floats (class,
+        bounding box, conf score). The bounding box will be located at indexes
+        1-4.
+        :type pred_y: list
+        :param true_y: ground truth detections, nested list of 6 floats (class,
+        bounding box, is crowded). The bounding box will be located at indexes
```

Comment:
That is my understanding! Since I was talking about true_y (gt) I used "is crowded". Let me know if I can clarify this more. 

But yes, that's confusing to me too :)